services/lockfile.ts

```diff
@@ -0,0 +1,57 @@
+import * as lockfile from "lockfile";
+import * as path from "path";
+
+export class LockFile implements ILockFile {
+
+	private get defaultLockFilePath(): string {
+		return path.join(this.$options.profileDir, "lockfile.lock");
+	}
+
+	private get defaultLockParams(): lockfile.Options {
```

Comment:
We could also `@cache()` this but I don't believe it necessary

Reply:
I agree it is not necessary here ;)